packages/api/src/controllers/asset.ts

```diff
@@ -74,18 +77,32 @@ function isPrivatePlaybackPolicy(playbackPolicy: PlaybackPolicy) {
   return true;
 }
 
-function defaultObjectStoreId(
-  { config, body }: Request,
+const secondaryStorageExperiment = "secondary_vod_storage";
+
+async function defaultObjectStoreId(
+  { config, body, user }: Request,
   isOldPipeline?: boolean
-): string {
+): Promise<string> {
   if (isOldPipeline) {
     return config.vodObjectStoreId;
   }
 
   if (isPrivatePlaybackPolicy(body.playbackPolicy)) {
     return config.vodCatalystPrivateAssetsObjectStoreId;
   }
-  return config.vodCatalystObjectStoreId || config.vodObjectStoreId;
+
+  const secondaryStorageEnabled = await isExperimentSubject(
+    secondaryStorageExperiment,
+    user.id
+  );
+  const secondaryjObjectStoreId = secondaryStorageEnabled
+    ? config.secondaryVodObjectStoreId
+    : undefined;
+  return (
+    secondaryjObjectStoreId ||
+    config.vodCatalystObjectStoreId ||
+    config.vodObjectStoreId
+  );
```

Comment:
```suggestion
const secondaryObjectStoreId = secondaryStorageEnabled && config.secondaryVodObjectStoreId;

return secondaryObjectStoreId || config.vodCatalystObjectStoreId || config.vodObjectStoreId;
```

Nit

Reply:
`yarn run prettier` is multilining everything unfortunately but the `&&` is perfect thanks